File: AILang/benchmarks/python/loop_hash.py
def loop_hash_bench(iterations: int) -> int:
    acc = 0
    i = 0
    while i < iterations:
        acc = acc + i
        if acc > 1000000000:
            acc = acc - 1000000000
        i += 1
    return acc


def main() -> int:
    iterations = 12_000_000
    result = loop_hash_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: AILang/benchmarks/rust/loop_hash.rs
fn loop_hash_bench(iterations: u32) -> u32 {
    let mut acc: u32 = 0u32;
    let mut i: u32 = 0;
    while i < iterations {
        acc += i;
        if acc > 1_000_000_000 {
            acc -= 1_000_000_000;
        }
        i += 1;
    }
    acc
}

fn main() {
    let iterations: u32 = 12_000_000;
    let result = loop_hash_bench(iterations);
    println!("{}", result);
}
